crates/cdk/src/wallet/proofs.rs

```diff
@@ -61,20 +77,55 @@ impl Wallet {
             .collect())
     }
 
+    /// Get this wallet's [Proofs] that match the args with transaction
+    pub(super) async fn get_proofs_with_tx(
```

Comment:
We have two apis for the same thing. Some times we have two fns one _with_tx. But then in other fns we just have one fn that accepts an Option<tx> we should be consistent with how we handle this.

Reply:
We will drop all options, any function that accepts a transaction is private / pub super